extensions/parquet/base/src/main/java/io/deephaven/parquet/base/ParquetFileReader.java

```diff
@@ -368,23 +358,23 @@ private static LogicalTypeAnnotation getLogicalTypeAnnotation(ConvertedType type
             case ENUM:
                 return LogicalTypeAnnotation.enumType();
             case DECIMAL:
-                int scale = schemaElement == null ? 0 : schemaElement.scale;
-                int precision = schemaElement == null ? 0 : schemaElement.precision;
+                final int scale = schemaElement == null ? 0 : schemaElement.scale;
+                final int precision = schemaElement == null ? 0 : schemaElement.precision;
                 return LogicalTypeAnnotation.decimalType(scale, precision);
             case DATE:
                 return LogicalTypeAnnotation.dateType();
             case TIME_MILLIS:
-                // TODO(deephaven-core#976) Assuming that time is adjusted to UTC
+                // isAdjustedToUTC parameter is ignored while reading Parquet TIME type, so assume it whatever
                 return LogicalTypeAnnotation.timeType(true, LogicalTypeAnnotation.TimeUnit.MILLIS);
             case TIME_MICROS:
-                // TODO(deephaven-core#976) Assuming that time is adjusted to UTC
                 return LogicalTypeAnnotation.timeType(true, LogicalTypeAnnotation.TimeUnit.MICROS);
             case TIMESTAMP_MILLIS:
-                // TODO(deephaven-core#976) Assuming that time is adjusted to UTC
-                return LogicalTypeAnnotation.timestampType(true, LogicalTypeAnnotation.TimeUnit.MILLIS);
+                // Converted type doesn't have isAdjustedToUTC parameter, so use the information from logical type
```

Comment:
This is an ugly hack that I had to do because of the code inside `ParquetFileReader::buildChildren` which looks like this:

```
if (schemaElement.isSetConverted_type()) {
    LogicalTypeAnnotation originalType =
            getLogicalTypeAnnotation(schemaElement.converted_type, schemaElement);
    LogicalTypeAnnotation newOriginalType = schemaElement.isSetLogicalType()
            && getLogicalTypeAnnotation(schemaElement.logicalType) != null
                    ? getLogicalTypeAnnotation(schemaElement.logicalType)
                    : null;
    if (!originalType.equals(newOriginalType)) {
        ((Types.Builder) childBuilder).as(originalType);
    }
}
```

Mainly we are trying to deduce the type of the column from `schemaElement.converted_type` and `schemaElement.logicalType` and in case of mismatch, we go with type deduced from `schemaElement.converted_type`.

The problem is that `schemaElement.converted_type` doesn't have any information about adjustments to UTC. So to deduce it correctly, I had to read the `isAdjustedToUTC` from the logical type.

Ideally, I would have liked to change the above code to prefer the type deduced from `schemaElement.logicalType` over `schemaElement.converted_type` because `converted_type` is deprecated and superseded by `logicalType` (based on the Javadoc for converted_type).
But this would be a much bigger change and impact all parquet types. So I added this hack which just impacts the TIMESTAMP type.

Reply:
What would be the side effect of switching to use the logical type?

Reply:
I have added a test commit [here](https://github.com/deephaven/deephaven-core/compare/main...malhotrashivam:deephaven-core:nightly/sm-test-branch) and started the [nightly jobs](https://github.com/malhotrashivam/deephaven-core/actions/runs/6827594756). Let me see how this work.

Reply:
The nightly jobs run fine.